lib/fs/fakefs.go

```diff
@@ -286,6 +306,10 @@ func (fs *fakefs) Mkdir(name string, perm FileMode) error {
 	fs.mut.Lock()
 	defer fs.mut.Unlock()
 
+	if fs.insens {
+		name = UnicodeLowercase(name)
```

Comment:
Same issue here, this will create an entry with a lowercase name.

Reply:
@imsodin thank you so much! I think I finally grok my blunder, or at least I'm on the path to do it. Will have to take several hours of excessive caffeine and nicotine intake to get it all straight (not likely to happen on business days), but I'll get there. Thanks again!

Reply:
That doesn't sound like much fun ;) From what I saw during my review I'd say you just need to go over the methods and ensure the following three things:

1. Create fake-entries with the name given as a parameter (not folded).
2. Once a fake-entry is created, no operation may change it's name (except for replacing it with a new one/renaming).
3. When stating change the name on a copy of the entry (should already be copied now as pointed out by Jakob) to be equal to the name given as a parameter.

I believe that should be doable without (overly much) intoxication :D

Reply:
Yep, I got the general steps already. What I meant by "to get _it all_ straight" was not the code itself (I think I'll straighten it up by tomorrow evening), but some misaligned (in my head) conceptions about how things work, so that I start speaking better Go, so that my future PRs don't result in your three wasting the whole lot of your time trying to talk sense to me. ;-)

Reply:
OK, I think it's good now. I fixed a couple of tests and wrote some new ones along the way, so that at least everything I've changed is covered. I might have missed something, though, so please do critisize! ;)